mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,617 @@
+from datetime import datetime
+from typing import List, Optional
+
+import weaviate
+import pandas as pd
+
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.integrations.libs.response import RESPONSE_TYPE
+from mindsdb.integrations.libs.response import HandlerResponse
+from mindsdb.integrations.libs.response import HandlerResponse as Response
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+from mindsdb.integrations.libs.vectordatabase_handler import (
+    FilterCondition,
+    FilterOperator,
+    TableField,
+    VectorStoreHandler,
+)
+from mindsdb.utilities import log
+from weaviate.util import generate_uuid5
+
+
+class WeaviateDBHandler(VectorStoreHandler):
+    """This handler handles connection and execution of the Weaviate statements."""
+
+    name = "weaviate"
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name)
+
+        self._connection_data = kwargs.get("connection_data")
+
+        self._client_config = {
+            "weaviate_url": self._connection_data.get("weaviate_url"),
+            "weaviate_api_key": self._connection_data.get("weaviate_api_key"),
+        }
+
+        if (
+            self._client_config["weaviate_url"] is None
+            or self._client_config["weaviate_api_key"] is None
+        ):
+            raise Exception(
+                "Both url + api key client secret are required for weaviate connection!"
+            )
+
+        self._client = None
+        self.is_connected = False
+        self.connect()
+
+    def _get_client(self) -> weaviate.Client:
+        client_config = self._client_config
+        if client_config is None:
+            raise Exception("Client config is not set!")
+
+        # decide the client type to be used, either persistent or httpclient
+        return weaviate.Client(
+            url=client_config["weaviate_url"],
+            auth_client_secret=weaviate.AuthApiKey(
+                api_key=client_config["weaviate_api_key"]
+            ),
+        )
+
+    def __del__(self):
+        if self._client is not None:
+            del self._client
+
+    def connect(self):
+        """Connect to a weaviate database."""
+        if self.is_connected:
+            return self._client
+
+        try:
+            self._client = self._get_client()
+            self.is_connected = True
+            return self._client
+        except Exception as e:
+            log.logger.error(f"Error connecting to weaviate client, {e}!")
+            self.is_connected = False
+
+    def disconnect(self):
+        """Close the database connection."""
+
+        if not self.is_connected:
+            return
+
+        self._client = None
+        self.is_connected = False
+
+    def check_connection(self):
+        """Check the connection to the Weaviate database."""
+        response_code = StatusResponse(False)
+        need_to_close = not self.is_connected
```

Comment:
rm additional variable

Reply:
resolved in b67cb1454fe8803bc304df7430cab6bcd5d670e7